internal/restinterface/externalhandler/externalhandler.go

```diff
@@ -431,11 +431,11 @@ func (h *Handler) APIV1RequestCloudSyncmgrPost(w http.ResponseWriter, r *http.Re
 		responseMsg = orchestrationapi.InvalidParameter
 		goto SEND_RESP
 	}
-	cloudsyncmgr.GetInstance().StartCloudSync(url)
+	cloudsyncmgr.GetInstance().StartCloudSync(url, appID)
```

Comment:
Can you define function names of cloudsyncmgr more properly?
`StartCloudSync` seems as if to run `cloudsyncmgr`, but it's not and the purpose of `StartCloudSync` at the moment is to run a mqtt client instead of cloudsyncmgr.
Also what about calling `mqttmgr.StartMQTTClient(host, clientID)` in the `RequestCloudSyncConf()`?

Reply:
What name would you suggest? We had initially thought of using startmqtt but thought in cloudsync mgr package this name as main function would sound confusing and hence we went with StartCloudSyncmgr function.

Reply:
If it were me, I don't make `StartCloudSync`. Perhaps I call `mqttmgr.StartMQTTClient` in the `RequestCloudSync`. I also recommend you to use [channels](https://go.dev/blog/concurrency-timeouts) instead of [for loop](https://github.com/lf-edge/edge-home-orchestration-go/blob/0589ede2610efbf9dd029d07a190ffe607d889d4/internal/restinterface/externalhandler/externalhandler.go#L435).

Reply:
Moved StartMQTTClient to RequestCloudSyncConf and used wiatgroup instead of sleep to check if the client creation is completed instead of for lopp
Done!